module/VuFindSearch/src/VuFindSearch/Feature/EventManagerTrait.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/**
+ * Trait to compose the event manager into a search class.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+
+namespace VuFindSearch\Feature;
+
+use Laminas\EventManager\EventManager;
+use Laminas\EventManager\EventManagerInterface;
+
+/**
+ * Trait to compose the event manager into a search class.
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+trait EventManagerTrait
+{
+    /**
+     * Event manager.
+     *
+     * @var EventManager
+     */
+    protected $events;
+
+    /**
+     * Set EventManager instance.
+     *
+     * @param EventManagerInterface $events Event manager
+     *
+     * @return void
+     */
+    public function setEventManager(EventManagerInterface $events)
+    {
+        // Using the class name as the event namespace is recommended; the
+        // VuFind\Search and VuFindSearch values are only retained for backward
+        // compatibility.
+        $events->setIdentifiers([self::class, 'VuFind\Search', 'VuFindSearch']);
```

Comment:
We need `\VuFindSearch\Service::class` here when called from Blender\Backend, but `self:class` results in `'VuFindSearch\Backend\Blender\Backend'`. That's causing the test failure, because Blender backend can't trigger the events for the correct identifier. How about something like `setSearchEventManager()` instead that would always use the three search-specific identifiers?

Reply:
(I'll push changes up after tests pass on my end).

Reply:
Works for me!